diff --git a/ssl/ssl_cert.c b/ssl/ssl_cert.c
index 126be66..014295a 100644
--- a/ssl/ssl_cert.c
+++ b/ssl/ssl_cert.c
@@ -863,6 +863,7 @@ int SSL_add_file_cert_subjects_to_stack(STACK_OF(X509_NAME) *stack,
     return ret;
 }
 
+#ifndef OPENSSL_NO_POSIX_IO
 int SSL_add_dir_cert_subjects_to_stack(STACK_OF(X509_NAME) *stack,
                                        const char *dir)
 {
@@ -918,6 +919,7 @@ int SSL_add_dir_cert_subjects_to_stack(STACK_OF(X509_NAME) *stack,
 
     return ret;
 }
+#endif
 
 static int add_uris_recursive(STACK_OF(X509_NAME) *stack,
                               const char *uri, int depth)
